modules/apps/27-interchain-accounts/controller/keeper/handshake.go

```diff
@@ -32,26 +34,25 @@ func (k Keeper) OnChanOpenInit(
 		return sdkerrors.Wrapf(channeltypes.ErrInvalidChannelOrdering, "expected %s channel, got %s", channeltypes.ORDERED, order)
 	}
 
-	connSequence, err := icatypes.ParseControllerConnSequence(portID)
-	if err != nil {
-		return sdkerrors.Wrapf(err, "expected format %s, got %s", icatypes.ControllerPortFormat, portID)
+	if !strings.HasPrefix(portID, icatypes.PortPrefix) {
+		return sdkerrors.Wrapf(porttypes.ErrInvalidPort, "controller port %s does not contain expected prefix %s", portID, icatypes.PortPrefix)
```

Comment:
maybe we should have ErrInvalidControllerPort and ErrInvalidHostPort and then use the expected, got format?

Reply:
yea. We should use "expected got" whenever possible

Reply:
Done! "expected, got format" did not register in my brain in the first comment 😄